ucrop/src/main/java/com/yalantis/ucrop/task/BitmapLoadTask.java

```diff
@@ -95,51 +102,72 @@ protected BitmapWorkerResult doInBackground(Void... params) {
             return new BitmapWorkerResult(e);
         }
 
-        final ParcelFileDescriptor parcelFileDescriptor;
-        try {
-            parcelFileDescriptor = mContext.getContentResolver().openFileDescriptor(mInputUri, "r");
-        } catch (FileNotFoundException e) {
-            return new BitmapWorkerResult(e);
-        }
+        Bitmap decodeSampledBitmap = null;
+        ContentResolver resolver = mContext.getContentResolver();
+        if (mBeforeAndroidQ) {
```

Comment:
It's unnecessary check, it's not really needed to use file descriptor. We fixed this in our fork, you can take a look

https://github.com/bandlab/uCrop/commit/6ffe481ea63cd0cc6df5d02cb5e859a3cfe773b5

Reply:
Thanks, I just considered Android10 compatibility. I looked at your fork and did a simple test, and it worked fine. Why is it closed?

Reply:
We just didn't have time to prepare proper PR, but we plan to do that. Also those changes changed behavior, image is not saved anymore to Downloads, it should be documented

Reply:
One more thing, to make a proper fix, you should send PR not to master, but to develop AND to feature/non_native branches, because non-native and native versions of the lib are essentially forks

Reply:
Yes, I found UCrop's jni in my project. I haven’t had time to modify it. I will close this pr and use your solution in my project.

Reply:
@joker-fu Keep in mind, that our fix is based on version 2.2.4 (non-native version)

Reply:
After several hours of incomplete testing, I came to the conclusion that your plan is more time-consuming. There are two main points: 1. CopyFile is executed regardless of the version, which increases the time 2. decodeStream is more time-consuming than decodeFileDescriptor

![image](https://user-images.githubusercontent.com/20105457/71660384-d3785380-2d85-11ea-8e0b-20759c0f8b05.png)

Reply:
This is the only correct way anyway. copyFile is required because it's used by other parts of the app. Using file descriptor is just incorrect

Reply:
Also, are you sure that your measurement is correct and you don't forget to include copyFile function, because it's required and for old and for new code

Reply:
I don't quite agree with FileDescriptor's incorrect statement (I want to know why), because UCrop was at least available before Android10. Then there is the issue of time. Even if you add the copy application time to the version that does not use copyFile, it still saves a little time. Of course, I only tested 2 phones.

Reply:
> UCrop was at least available before Android10
I don't say that it will not work, it's just too hacky, and broken work with Android 10 proofs this. ContentResolve is the only correct way to do that